pkgs/tools/X11/xborders/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+  , buildPythonPackage
+  , python3Packages
+  , fetchFromGitHub
+  , libwnck
+  , gtk3
+  , libnotify
+  , wrapGAppsHook
+  , gobject-introspection
+}:
+
+buildPythonPackage rec {
+  pname = "xborders";
+  version = "3.4"; # in version.txt
+
+  src = fetchFromGitHub {
+    owner = "deter0";
+    repo = "xborder";
+    rev = "8a031736c9e89a031428f33adcd42da7512d558e";
+    hash = "sha256-Q2WWXekUL/OOr/OBqJOZbAHaWURbP+coBVp206uPSNY=";
+  };
+
+  buildInputs = [
+    libwnck
+    gtk3
+    libnotify
+  ];
+
+  nativeBuildInputs = [
+    wrapGAppsHook
+    gobject-introspection
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    pycairo
+    requests
+    pygobject3
+  ];
+
+  postPatch = ''
+    cat >setup.py <<'EOF'
```

Comment:
Why is this file needed to begin with?

I would ask if how the other distros are packaging this, but I have found none.

Supposing the file is needed:

1. Open a shell;
2. `cd local-nixpkgs-directory/pkgs/tools/X11/xborders`;
3. `nano config.py` -- insert contents to it as needed
4. `nano default.nix` -- insert something like

```nix
postPatch = ''
  ln -s ${./setup.py} setup.py
'';
```

Reply:
Ah alright, I was thinking one wanted to include `setup.py` as an extra source rather than just being linked in with a script. In terms of getting variable substitution work for the description, etc. I guess one can use `substituteAll`?

`setup.py` is required for `buildPythonPackage` (it fails without it), and the documentation for fixing a missing `setup.py` included shows including it in `preBuild`, which is why I had it that way originally: [Fix Missing `setup.py`](https://nixos.wiki/wiki/Packaging/Python#Fix_Missing_setup.py)

Reply:
Understood, but not much.

The arguments and example from the wiki are cogent but they lack elegance.
It is better to create a separate file `setup.py` and move it instead of embedding it in a heredoc in a shell script.

About using `substitute`, I believe this is ugly and unneeded.

Reply:
Alright. I agree that using `substituteAll` is ugly but I feel like having two update two files for a version bump would be worse. I've added the changes.